Change 'attach_keyword' option from string to regex

weisslj opened the pull request:
A regular expression is much more useful than a single word to check for forgotten attachments, especially when writing in multiple languages.

This is inspired by the [muttng patchset abort_noattach](https://github.com/codito/muttng-patchset/blob/master/patches/muttng.abort_noattach.diff), which I use for years, but newly implemented to make minimal changes to the neomutt source code.

The keyword is no longer mentioned in the error message as regular expressions tend to be longer.

The option 'attach_keyword' could be renamed 'attach_remind_regex' as in the muttng patch.

This pull request would solve #138.

flatcap commented:
Hi @weisslj 
Thank you very much for this.

The code looks good.
Please could you add a few regex examples to the docs.
Also, feel free to add yourself to the credits section :-)

weisslj commented:
Thanks for the review and also for inviting me to the team! I added a regex example and added me to the credits section. I also created a devel/abort_noattach_regexp feature branch for this.

P-EB commented:
I'd suggest to add a "default" regex in the Muttrc default configuration file.

Maybe there'd be a way to handle translations of this regex for other languages, I don't know. :)

flatcap commented:
We should provide examples for all the languages we speak:

English: `<attach(|ed|ments?)>` matches whole words:
- attach
- attached
- attachment
- attachments

flatcap commented:
I've just searched the code and we should keep all the regexes in lower case.

init.h defines the variable as DT_RX
```
{ "attach_keyword",  DT_RX,  R_NONE, UL &AttachKeyword, UL "attach" },
```
At startup, we get this stack:
- `main()`
- `mutt_init()`
- `mutt_restore_default()`

For a DT_RX type, it checks if there are any upper case letters `mutt_which_case()`
If there are none, then the regex will ignore the case of the strings: `REG_ICASE`

P-EB commented:
French: `<joint(|e?)s?>` matches whole words:
- joint
- joints
- jointe
- jointes

weisslj commented:
Thanks for the suggestions and examples! I agree that a default regex would be good. Since we cannot provide it for "all" languages I suggest we stick to English (@flatcap's example but with correct `<` and `>` escaping):

```
\\<attach(|ed|ments?)\\>
```

Also good find with the regex case sensitivity, it is even mentioned in the manual: http://www.mutt.org/doc/manual/#regexp

The new patch is 22d3219.

weisslj commented:
I won't be on my computer the next three weeks.

flatcap commented:
OK, thanks for letting me know.
Have a nice holiday :-)

flatcap commented:
Merged in 337f6e3.  Thanks.

weisslj commented:
Thanks @flatcap!
